think_reason_learn/policy_induction/_policy_induction.py

```diff
@@ -0,0 +1,1377 @@
+
+            if not isinstance(config.Cs, (list, tuple, np.ndarray)):
+                raise ValueError("Cs must be an iterable of floats")
+
+            return config
+
+        raise ValueError(
+            f"config must be a WeightTrainerConfig, dict, "
+            f"or None, but got {type(cfg).__name__}"
+        )
+
+    def _set_save_path(self, save_path: str | PathLike[str] | None) -> Path:
+        if save_path is None:
+            return (Path(os.getcwd()) / "policy_induction" / self.name).resolve()
+        else:
+            save_path = Path(save_path).resolve()
+            if save_path.is_file():
+                raise ValueError("Please provide a directory, not a file.")
+            return save_path
+
+    def _set_initial_memory_df(self) -> pd.DataFrame:
+        """Initialize the policy memory DataFrame with explicit dtypes.
+
+        Returns:
+            pd.DataFrame: A DataFrame with the following columns:
+                - policy (str): The text of the policy.
+                - predictions (object): The corresponding binary predictions
+                  for each sample in X (keeps positional alignment with X).
+        """
+        return pd.DataFrame(
+            {
+                "policy": pd.Series([], dtype=str),
+                "predictions": pd.Series([], dtype=object),
+            },
+        )
+
+    def get_memory(self) -> pd.DataFrame:
+        """Get memory dataframe.
+
+        Returns:
+            pd.DataFrame: A DataFrame with the following columns:
+                - policy (str): The text of the policy.
+                - predictions (object): The corresponding binary predictions
+                  for each sample in X (keeps positional alignment with X).
+                - weight (float): The weight assigned to this policy, default 0.
+        """
+        return self._policy_memory
+
+    async def set_task(
+        self,
+        task_description: str | None = None,
+        instructions_template: str | None = None,
+    ) -> str:
+        """Initialize policy generation instructions template.
+
+        This sets the task description for Policy Induction.
+        Either sets a custom template or generates one from task description using LLM.
+        For most users, LLM generation is recommended over custom templates.
+
+        Args:
+            instructions_template: Custom template to use. Must contain
+                '<max_policy_length>' tag. If None, generates template
+                from task_description using LLM.
+            task_description: Description of classification task to help LLM generate
+                the template.
+
+        Returns:
+            The policy generation instructions template.
+
+        Raises:
+            ValueError: If template missing required tag or generation fails.
+            AssertionError: If both parameters are None.
+        """
+        assert task_description is not None, "Task_description must be provided"
```

Comment:
when `instructions_template` is provided, then `task_description` can be ignore. You should therefore assert at least one of them. It's currently only `the task_description`